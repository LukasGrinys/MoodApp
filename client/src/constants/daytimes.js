export const daytimes = {
    morning: 'Morning',
    afternoon: 'Afternoon',
    evening: 'Evening',
    night: 'Night'
}