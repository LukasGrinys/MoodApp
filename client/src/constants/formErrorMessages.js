export const formErrorMessages = {
    required : 'Required field',
    email: 'Enter a proper e-mail address'
}