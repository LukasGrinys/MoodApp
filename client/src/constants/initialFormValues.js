export const initialFormValues = {
    email: '',
    password: '',
    firstName: '',
    lastName: '',
    confirmPassword: '',
    currentPassword: ''
}