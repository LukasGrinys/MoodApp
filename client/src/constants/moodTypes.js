const moodTypes = [
    {
        above: 7.5,
        below: 10,
        text: 'Great'
    },
    {
        above: 5,
        below: 7.5,
        text: 'Good'
    },
    {
        above: 3,
        below: 5,
        text: 'Average'
    },
    {
        above: 0,
        below: 3,
        text: 'Bad'
    }
];

export default moodTypes;