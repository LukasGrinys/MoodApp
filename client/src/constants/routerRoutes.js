export const routerRoutes = {
    home: '/',
    login: '/login',
    logout: '/logout',
    dashboard: '/dashboard',
    addNewLog: '/addlog',
    register: '/register',
    settings: '/settings'
}