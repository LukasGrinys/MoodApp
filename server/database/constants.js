const COLLECTION_NAMES = {
    users : 'users',
    logs : 'logs'
}

module.exports = {
    COLLECTION_NAMES
}