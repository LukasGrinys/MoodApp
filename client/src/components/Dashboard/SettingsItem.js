import React from 'react';
import { Link } from 'react-router-dom';
import parentClasses from './Dashboard.module.scss';
import styles from './SettingsItem.module.scss';
import { routerRoutes } from '../../constants/routerRoutes';
import { useTheme } from '../../contexts/ThemeContext';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCog } from '@fortawesome/free-solid-svg-icons';

import classNames from 'classnames';

const SettingsItem = () => {
    const darkTheme = useTheme();

    return (
        <div className={classNames(
            parentClasses.gridItem,
            parentClasses.shrink
            )}>
            <h3 className={parentClasses.gridItemHeader}>Settings</h3>
            <div className={parentClasses.flexCenter}>
                <div className={styles.settingsIconWrapper}>
                    <Link to={routerRoutes.settings}>
                        {}
                        <FontAwesomeIcon icon={faCog} className={classNames(
                            styles.settingsIcon,
                            darkTheme && styles.dark
                            )}/>
                    </Link>
                </div>    
            </div>
        </div>
    );
};

export default SettingsItem;