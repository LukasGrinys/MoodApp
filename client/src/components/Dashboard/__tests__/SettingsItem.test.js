import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import SettingsItem from '../SettingsItem';
import { mount } from 'enzyme';
import { routerRoutes } from '../../../constants/routerRoutes';

const wrapper = mount(
    <Router>
        <SettingsItem/>
    </Router>
)

test('Renders properly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Has settings icon (Cog)', () => {
    const icon = wrapper.find('FontAwesome');

    expect(icon.length).toBe(1);
    expect(icon.props()).toEqual({
        className: 'settingsIcon',
        name: 'cog'
    })
});

test('Has a link to settings page', () => {
    const link = wrapper.find('Link');
    
    expect(link.length).toBe(1);
    expect(link.prop('to')).toBe(routerRoutes.settings);
})