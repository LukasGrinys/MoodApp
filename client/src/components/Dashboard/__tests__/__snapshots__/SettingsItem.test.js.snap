// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <SettingsItem>
      <div
        className="gridItem shrink"
      >
        <h3
          className="gridItemHeader"
        >
          Settings
        </h3>
        <div
          className="flexCenter"
        >
          <div
            className="settingsIconWrapper"
          >
            <Link
              to="/settings"
            >
              <LinkAnchor
                href="/settings"
                navigate={[Function]}
              >
                <a
                  href="/settings"
                  onClick={[Function]}
                >
                  <FontAwesome
                    className="settingsIcon"
                    name="cog"
                  >
                    <span
                      aria-hidden={true}
                      className="fa fa-cog settingsIcon"
                    />
                  </FontAwesome>
                </a>
              </LinkAnchor>
            </Link>
          </div>
        </div>
      </div>
    </SettingsItem>
  </Router>
</BrowserRouter>
`;
